src/platform/core/src/device/ogl_video_device.cpp

```diff
@@ -202,6 +203,16 @@ void OGLVideoDevice::CreateShaderPrograms() {
     }
   }
 
+  //LCD grid pass
+  if(config->video.lcd_grid) {

```

Comment:
You're pushing the shader pass at the end of the chain. This breaks certain assumptions in the current code regarding when each pass happens, which leads to artefacts when ghosting effect or xBRZ are enabled, and also renders bilinear sharp shader as regular bilinear.

Can you please describe the expected output of this shader, and expected interactions with other present shaders? Right now, my understanding is this should be used as a final pass with integer scaling enabled, although it theoretically can be overlaid on top of any output.

Also, please adjust formatting: add a space to the comment line, so it reads `// LCD...`; remove the empty line after `if(success) {`; and leave only one empty line before `UpdateShaderUniforms();` call.

Reply:
Hello, sorry for the late answer. 
The objective of the shader is to draw borders on each of the "GBA pixels".
In theory it can be applied with any scaling, but the pixels may not match the grid. 
The "best setup" in theory is integer scaling ON and nearest filter.
If I undestand the problem. It's needed to change the order of the shaders. 
I tried to make "a correction" like in xbrz with ghosting, but it didn't work, xbrz is disabled and If I turn off lcd grid, the image is flipped.
This is harder than I imagined. hehehehe.

Reply:
And thank you for all your help.

Reply:
Honestly, I'd probably just slap it alongside the rest of output shaders and revert back to its original name (so lcd1x instead of "LCD grid"). By default, nearest interpolation is performed, so the user only has to turn on integer scaling. Then you don't have to worry about breaking anything.

Adding more passes will sooner or later necessitate a more advanced rendering loop, at which point it'll probably make sense to adopt RetroArch's Slang shader format and just leave passes configuration to the user, or something like that. I'd certainly prefer to leave the current simple and curated experience with a minimal set of shaders arranged in a specific way ~~because it's less work~~.